data-prepper-core/src/main/java/org/opensearch/dataprepper/sourcecoordination/LeaseBasedSourceCoordinator.java

```diff
@@ -450,6 +439,6 @@ private void giveUpAndSaveGlobalStateForPartitionCreation(final SourcePartitionS
     private SourcePartitionStoreItem getItemWithAction(final String partitionKey, final String action, final Boolean fromAcknowledgmentsCallback) {
         // The validation against activePartition in partition manager needs to be skipped when called from acknowledgments callback
         // because otherwise it will fail the validation since it is actively working on a different partition when ack is received
-        return fromAcknowledgmentsCallback ? getSourcePartitionStoreItem(partitionKey, action) : validateAndGetSourcePartitionStoreItem(partitionKey, action);
+        return fromAcknowledgmentsCallback ? getSourcePartitionStoreItem(partitionKey, action) : getSourcePartitionStoreItem(partitionKey, action);
```

Comment:
Nit: No need for this ternary operator anymore.

Reply:
Oh, right. Will change it.